platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/connectivity/MqttGenerator.xtend

```diff
@@ -68,64 +82,245 @@ class MqttGenerator extends AbstractSystemResourceGenerator {
 		
 		static const uint16_t MQTT_BROKER_PORT = «brokerPort»;
 		
-		MqttSession_T mqttSession;
+		/**<  Macro for the non secure serval stack expected MQTT URL format */
+		#define MQTT_URL_FORMAT_NON_SECURE          "mqtt://%s:%d"
+		
+		/**
+		 * @brief   Structure to represent the MQTT connect features.
+		 */
+		struct ConnectivityMQTT_Connect_S
+		{
+			const char * ClientId; /**< The client identifier which is a identifier of each MQTT client connecting to a MQTT broker. It needs to be unique for the broker to know the state of the client. */
+			const char * BrokerURL; /**< The URL pointing to the MQTT broker */
+			uint16_t BrokerPort; /**< The port number of the MQTT broker */
+			bool CleanSession; /**< The clean session flag indicates to the broker whether the client wants to establish a clean session or a persistent session where all subscriptions and messages (QoS 1 & 2) are stored for the client. */
+			uint32_t KeepAliveInterval; /**< The keep alive interval (in seconds) is the time the client commits to for when sending regular pings to the broker. The broker responds to the pings enabling both sides to determine if the other one is still alive and reachable */
+		};
+		
+		/**
+		 * @brief   Typedef to represent the MQTT connect feature.
+		 */
+		typedef struct ConnectivityMQTT_Connect_S ConnectivityMQTT_Connect_T;
+		
+		/**
+		 * @brief   Structure to represent the MQTT publish features.
+		 */
+		struct ConnectivityMQTT_Publish_S
+		{
+			const char * Topic; /**< The MQTT topic to which the messages are to be published */
+			uint32_t QoS; /**< The MQTT Quality of Service level. If 0, the message is send in a fire and forget way and it will arrive at most once. If 1 Message reception is acknowledged by the other side, retransmission could occur. */
+			const char * Payload; /**< Pointer to the payload to be published */
+			uint32_t PayloadLength; /**< Length of the payload to be published */
+		};
+		
+		/**
+		 * @brief   Typedef to represent the MQTT publish feature.
+		 */
+		typedef struct ConnectivityMQTT_Publish_S ConnectivityMQTT_Publish_T;
+		
+		/**< Handle for MQTT subscribe operation  */
+		static SemaphoreHandle_t MqttSubscribeHandle;
+		/**< Handle for MQTT publish operation  */
+		static SemaphoreHandle_t MqttPublishHandle;
+		/**< Handle for MQTT send operation  */
+		static SemaphoreHandle_t MqttSendHandle;
+		/**< Handle for MQTT send operation  */
+		static SemaphoreHandle_t MqttConnectHandle;
+		/**< MQTT session instance */
+		static MqttSession_T MqttSession;
+		/**< MQTT connection status */
+		static bool MqttConnectionStatus = false;
```

Comment:
What is this status? Because it's boolean it's hard to understand the semantics of this flag. The "documentation" isn't helping either, I'm afraid. Is it connected, authenticated, secured, available?

Please rename to something like `MqttIsConnected` (depending on what kind of status this is). The same goes for the "status" variables below.